### DISCUSSION TITLE
Specifying externalhosts multiple time in lnd.conf

### DISCUSSION THREAD

**Discussion by johnheenan** [lightningnetwork/lnd#9071] [Q&A] [answered]
From https://github.com/lightningnetwork/lnd/blob/31a6f87c4d0c2c1b095e0e909b113da29ea7dc9f/sample-lnd.conf#L77C1-L84C39

```
; A list of domains for lnd to periodically resolve, and advertise the resolved
; IPs for the backing node. This is useful for users that only have a dynamic IP,
; or want to expose the node at a domain.
; Default:
;   externalhosts=
; Example (option can be specified multiple times):
;   externalhosts=my-node-domain.com
;   externalhosts=my-second-domain.com
```
Assume a DNS lookup returns a single different IP address from each of two specified externalhosts.

Is the purpose to specify to a peer that the LND server can be reached on at least one of the IP addresses at any time but not necessarily both at the same time?

Or is the purpose to offer a peer a choice as to which IP address to use?

**Comment by guggero** [comment, marked as answer]
The purpose of the `externalhost` config option is mostly to allow users with dynamic IP addresses on their machines to be reachable. The reason there can be multiple domains likely just is for uniformity with other similar configuration options that allow multiple entries. I can't come up with a use case where this would be useful from the top of my head.

But generally the p2p protocol for gossiping a node's addresses (currently) only allows _IP_ addresses to be sent, not domain names. And someone trying to reach your node will just try any of the addresses until a connection is successful.
So your node _can_ just be reachable on one of the addresses it advertises, but that will make things less smooth because a random attempt at any address that currently doesn't work will first need to time out.

**Reply by johnheenan** [reply]
Thanks. Very helpful.
